### PR TITLE
Added geqrf

### PR DESCRIPTION
**Pull request by tfalders**
Adds geqrf to hipSOLVER

### CODE REVIEW

**Review by cgmb — APPROVED** [review]
LGTM

File: clients/include/hipsolver.hpp [hunk 1]
```diff
@@ -51,6 +51,150 @@ inline testMarshal_t bool2marshal(bool FORTRAN, bool STRIDED, bool ALT = false)
...
+                                         int               lda,
+                                         int               stA,
+                                         float*            tau,
+                                         int               stT,
+                                         float*            work,
+                                         int               lwork,
+                                         int*              info,
+                                         int               bc)
+{
+    switch(bool2marshal(FORTRAN, bc != 1))
```
[the remark this answers is not in the document]
Comment (jzuniga-amd) [reply clients/include/hipsolver.hpp:111]: Yes, that is what I saw with potrf_batched...
There is one stridedbatched routine in cusolver, but I don't know if we will have an equivalent in rocsolver. I think we can safely remove this condition, but not a big deal ...

**Comment by tfalders** [comment]
My guess is that the CI is failing because of an issue with the workspace management. I've revised it, but since I can't reproduce the CI error, I'm going to try pushing the change to see if that fixes the failures. If it does, I'll re-request a review from @cgmb, since it's a fairly big change; otherwise, I'll force remove the change.

File: library/src/hcc_detail/hipsolver.cpp [hunk 2]
```diff
@@ -147,6 +147,201 @@ catch(...)
...
+                                  float*            tau,
+                                  float*            work,
+                                  int               lwork,
+                                  int*              devInfo)
+try
+{
+    if(work != nullptr)
+        rocblas_set_workspace((rocblas_handle)handle, work, lwork);
+    else
+    {
```
[the remark this answers is not in the document]
Comment (jzuniga-amd) [reply library/src/hcc_detail/hipsolver.cpp:249]: Sounds good! 
Just please add to the documentation that when providing a nullptr workspace, if the backend is rocsolver, the library will allocate it automatically.

**Review by jzuniga-amd — APPROVED** [review]
Looks good.

File: clients/gtest/CMakeLists.txt [hunk 3]
```diff
@@ -22,9 +22,9 @@ include_directories(${GTEST_INCLUDE_DIRS})
 
 set(hipsolver_test_source
   hipsolver_gtest_main.cpp
-  geqrf_gtest.cpp
-  getrf_gtest.cpp
   getrs_gtest.cpp
+  getrf_gtest.cpp
+  geqrf_gtest.cpp
```
Comment (cgmb) [inline clients/gtest/CMakeLists.txt]: What problem did this fix?

Comment (tfalders) [reply clients/gtest/CMakeLists.txt]: For some reason getrs has been having memory access faults on CI when using that original test ordering. Juan said we had a similar issue in rocSOLVER before, and he solved it by reordering the tests. Apparently there may be a mystery bug in our client code (e.g. device_vectors).

**Pull request merged**